opendatahub-io/odh-dashboard
# Refactor model serving section on the project details page

Closes #1941
Closes #1942 
Closes #1944 

## Description


This PR does the following things:
1. When both KServe and ModelMesh are enabled, and user creates a project, in the model serving section, we show them 2 cards

<img width="1728" alt="Screenshot 2023-10-26 at 3 50 21 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/309a1531-da7d-409e-8ff8-b334304d3718">

2. The user could click on either card footer button to open the corresponding modal based on their choice. We only show multi-model serving platform clicking behavior here because the single model serving platform modal will be implemented in #1943 

<img width="1728" alt="Screenshot 2023-10-26 at 3 51 09 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/44c762e7-5df1-4a9e-b707-7a0787d3cbba">

3. When the project is not labeled, but we only enabled either KServe or ModelMesh, we should the corresponding table with the empty state
4. When the project is labeled, directly show the labeled corresponding table (We add a tag at the top right of the section to indicate the enabled platform)

<img width="1728" alt="Screenshot 2023-10-25 at 11 22 09 AM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/44cc7744-5b8c-485a-bf7f-4dd507ed1f30">

## How Has This Been Tested?



1. Go to check the old projects, see if they are `Multi-model serving enabled` because they should be (We don't have KServe before and all the projects should have `modelmesh-enabled` label)
2. Go to create a new project
3. Try to choose different platform, see the change in the title action button
4. The single model platform button is not working now because we need the KServe modal. So try to choose multi-model serving platform and add a server
5. Once you successfully add a server, the project should be labeled and you can never see the selection
7. Try to go to the dashboard config to disable either KServe or ModelMesh
8. Wait for 2 mins, and create a new project, make sure this time you don't see the selection card, but it will directly take you to the empty table view based on the feature you enabled

## Test Impact


Create 2 stories for future testing, will add more integration tests when the KServe modal is implemented
Add unit test for `getModelServingPlatform` function

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
